### PR TITLE
Expose build info via prometheus metrics

### PR DESCRIPTION
**Pull request by palourde**
## What is this change?

It adds the Sensu Go build info to the prometheus metrics, so it can be easily exposed via Grafana.

## Why is this change necessary?

Closes https://github.com/sensu/sensu-perf/issues/23

## Does your change need a Changelog entry?

Added

## Do you need clarification on anything?

Nope

## Were there any complications while making this change?

Nope

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Nope

## How did you verify this change?

Manually deployed a build from this branch to the SPDC.

## Is this change a patch?

Nope

### CODE REVIEW

File: version/version.go [hunk 1]
```diff
@@ -33,8 +35,21 @@ var (
...
 	// (e.g. go1.14.2)
 	GoVersion string = runtime.Version()
+
+	promBuildInfo = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "sensu_go_build_info",
+			Help: "Sensu Go build information",
+		},
+		[]string{"version", "builddate", "buildsha", "goversion"},
+	).WithLabelValues(Version, BuildDate, BuildSHA, GoVersion)
```
Comment (ccressent) [inline version/version.go:45]: I'm not sure that the build date is that valuable or useful; in fact there are situations where it's actively harmful.

Nothing tells you that the clock of the building machine was correctly set, or that it didn't massively change between 2 builds, making the comparison of build date/time unreliable in general.

More importantly, this single nugget of information makes the build not easily reproducible: if you build, checksum the resulting binary, build and checksum again, you'll get 2 different checksums because time is made an implicit build input. This causes problems for some systems that try to deal with software in a reproducible fashion, e.g. Nix/NixOS (the sensu-go "package" there rebuilds from source and scrubs the build date for that reason).

TL;DR: I think we should drop `BuildDate`.

File: version/version.go [hunk 2]
```diff
@@ -33,8 +35,21 @@ var (
...
 )
 
+func init() {
+	prometheus.MustRegister(promBuildInfo)
+	promBuildInfo.Set(1)
```
Comment (ccressent) [inline version/version.go:50]: I don't understand why `.Set(1)` in necessary; what is it for?

**Review by ccressent — APPROVED** [review]

File: version/version.go (hunk 2 above)
Comment (palourde) [reply version/version.go:50]: It's required so the `sensu_go_build_info` metric has a value (it could have been any integer value really), otherwise it will not show up in the metrics endpoint
